tactics/hints.ml

```diff
@@ -669,7 +677,20 @@ let try_head_pattern c =
 
 let with_uid c = { obj = c; uid = fresh_key () }
 
+let secvars_of_idset s =
+  Id.Set.fold (fun id p ->
+      if is_section_variable id then
+        Id.Pred.add id p
+      else p) s Id.Pred.empty
+
```

Comment:
Is `is_section_variable` robust enough? If a goal variable takes the place of a section variable with same name, will the test be able to notice it?